packages/lib/src/vis/line/Glyphs.tsx

```diff
@@ -0,0 +1,56 @@
+import type { NumArray } from '@h5web/shared';
+import type { PointsProps } from '@react-three/fiber';
+
+import { useGeometry } from '../hooks';
+import { useVisCanvasContext } from '../shared/VisCanvasProvider';
+import { hasR3FEventHandlers } from '../utils';
+import GlyphMaterial from './GlyphMaterial';
+import GlyphsGeometry from './glyphsGeometry';
+import { GlyphType } from './models';
+
+interface Props extends PointsProps {
+  abscissas: NumArray;
+  ordinates: NumArray;
+  glyphType?: GlyphType;
+  color: string;
+  size?: number;
+  visible?: boolean;
+  ignoreValue?: (val: number) => boolean;
+}
+
+function Glyphs(props: Props) {
+  const {
+    abscissas,
+    ordinates,
+    glyphType = GlyphType.Cross,
+    color,
+    size = 6,
+    visible = true,
+    ignoreValue,
+    ...pointsProps
+  } = props;
+
+  const { abscissaScale, ordinateScale } = useVisCanvasContext();
+
+  const geometry = useGeometry(
+    GlyphsGeometry,
+    ordinates.length,
+    visible && {
+      abscissas,
+      ordinates,
+      abscissaScale,
+      ordinateScale,
+      ignoreValue,
+    },
+    hasR3FEventHandlers(pointsProps),
```

Comment:
Is this really needed ? I guess it would not be a huge performance dent to recompute the bounding sphere even if it is not interactive.

Reply:
Hehe, unfortunately it is. Reproduction:

1. Comment out this line
2. Start Storybook and expand the sidebar to fill half the screen
3. Navigate to the _DataCurve > Interactive_ story — notice that all points are interactive
4. Shrink the sidebar back to its original size — notice that the points at the start and end of the curve are no longer interactive.

What happens is, when the bounding sphere is not computed, Three computes it automatically on first paint. Then, when we update the geometry and repaint, the bounding sphere still exists so Three doesn't know it needs to be recomputed.

For a bit more context: when an interaction occurs, Three checks to see which objects on the scene interacts with the "ray". To speed this up, it first checks if the ray intersects the bounding sphere of each object's geometry. If the bounding sphere doesn't cover the whole geometry and a ray occurs outside of it, then the raycaster will discard the object as a potential intersection candidate very quickly.

Reply:
The above is one solution though. The advantage is that it just works for components like `Line` and `Glyphs`; the user doesn't have to know about `computeBoundingSphere` at all. The downside is that it's a bit magical/implicit.

Another option would be to let users call `computeBoundingSphere` themselves, perhaps by providing an `onAfterUpdate` hook (otherwise, they'd have to write a `useLayoutEffect` with the same dependencies as inside `useGeometry`...)